ros/ros/ros2.js

```diff
@@ -133,52 +142,89 @@ class ROS2 {
     return list.find(({name}) => name == service)?.types[0];
   }
 
+  _destroySubscriber(subscriber){
+    if (subscriber && !subscriber.__destroyed){
+      subscriber.__destroyed = true;
+      this.node.destroySubscription(subscriber);
+    }
+  }
   /** Subscribe to the named topic of the named type. Each time a new message
   * is received the provided callback is called. For available options see
   * https://robotwebtools.github.io/rclnodejs/docs/0.22.3/Node.html#createSubscription.
   * The default `options.qos.reliability` is best-effort.
+  * options object can contain: "throttleMs": throttle-in-milliseconds.
   * */
   subscribe(topic, type, onMessage, options = {}) {
     this.requireInit();
+    let firstLatchedMessage;
     const ros2Type = toROS2Type(type);
-    const sub = this.node.createSubscription(
-      ros2Type, topic, {qos, ...options}, onMessage);
-    this.subscriptions[topic] = sub;
+    const _onMessage = options?.throttleMs ?
+      _.throttle(onMessage, options.throttleMs) :
+      onMessage;
+
```

Comment:
We should add a comment here describing the approach and why we are doing this (to support latching even when we don't know whether the publisher is publishing transient-local or not).